rpc/transaction_test.go

```diff
@@ -23,54 +23,31 @@ func TestTransactionByHash(t *testing.T) {
-		Version: TransactionV1,
-		Nonce:   utils.TestHexToFelt(t, "0x2d"),
+	var DeclareTxnV2Example = DeclareTxnV2{
+		Type:              TransactionType_Declare,
+		Version:           TransactionV2,
+		MaxFee:            utils.TestHexToFelt(t, "0x4a0fbb2d7a43"),
+		ClassHash:         utils.TestHexToFelt(t, "0x79b7ec8fdf40a4ff6ed47123049dfe36b5c02db93aa77832682344775ef70c6"),
+		CompiledClassHash: utils.TestHexToFelt(t, "0x7130f75fc2f1400813d1e96ea7ebee334b568a87b645a62aade0eb2fa2cf252"),
+		Nonce:             utils.TestHexToFelt(t, "0x16e"),
 		Signature: []*felt.Felt{
-			utils.TestHexToFelt(t, "0xe500c4014c055c3304d8a125cfef44638ffa5b0f6840916049667a4c38aa1c"),
-			utils.TestHexToFelt(t, "0x45ac538bfce5d8c5741b4421bbdc99f5849451acae75d2048d7cc4bb029ca77"),
-		},
-		SenderAddress: utils.TestHexToFelt(t, "0x66dd340c03b6b7866fa7bb4bb91cc9e9c2a8eedc321985f334fd55de5e4e071"),
-		Calldata: []*felt.Felt{
-			utils.TestHexToFelt(t, "0x3"),
-			utils.TestHexToFelt(t, "0x39a04b968d794fb076b0fbb146c12b48a23aa785e3d2e5be1982161f7536218"),
-			utils.TestHexToFelt(t, "0x2f0b3c5710379609eb5495f1ecd348cb28167711b73609fe565a72734550354"),
-			utils.TestHexToFelt(t, "0x0"),
-			utils.TestHexToFelt(t, "0x3"),
-			utils.TestHexToFelt(t, "0x3207980cd08767c9310d197c38b1a58b2a9bceb61dd9a99f51b407798702991"),
-			utils.TestHexToFelt(t, "0x2f0b3c5710379609eb5495f1ecd348cb28167711b73609fe565a72734550354"),
-			utils.TestHexToFelt(t, "0x3"),
-			utils.TestHexToFelt(t, "0x3"),
-			utils.TestHexToFelt(t, "0x42969068f9e84e9bf1c7bb6eb627455287e58f866ba39e45b123f9656aed5e9"),
-			utils.TestHexToFelt(t, "0x2f0b3c5710379609eb5495f1ecd348cb28167711b73609fe565a72734550354"),
-			utils.TestHexToFelt(t, "0x6"),
-			utils.TestHexToFelt(t, "0x3"),
-			utils.TestHexToFelt(t, "0x9"),
-			utils.TestHexToFelt(t, "0x47487560da4c5c5755897e527a5fda37422b5ba02a2aba1ca3ce2b24dfd142e"),
-			utils.TestHexToFelt(t, "0xde0b6b3a7640000"),
-			utils.TestHexToFelt(t, "0x0"),
-			utils.TestHexToFelt(t, "0x47487560da4c5c5755897e527a5fda37422b5ba02a2aba1ca3ce2b24dfd142e"),
-			utils.TestHexToFelt(t, "0x10f0cf064dd59200000"),
-			utils.TestHexToFelt(t, "0x0"),
-			utils.TestHexToFelt(t, "0x47487560da4c5c5755897e527a5fda37422b5ba02a2aba1ca3ce2b24dfd142e"),
-			utils.TestHexToFelt(t, "0x21e19e0c9bab2400000"),
-			utils.TestHexToFelt(t, "0x0"),
+			utils.TestHexToFelt(t, "0x5569787df42fece1184537b0d480900a403386355b9d6a59e7c7a7e758287f0"),
+			utils.TestHexToFelt(t, "0x2acaeea2e0817da33ed5dbeec295b0177819b5a5a50b0a669e6eecd88e42e92"),
 		},
+		SenderAddress: utils.TestHexToFelt(t, "0x5fd4befee268bf6880f955875cbed3ade8346b1f1e149cc87b317e62b6db569"),
 	}
 
 	testSet := map[string][]testSetType{
 		"mock": {
 			{
-				TxHash:      utils.TestHexToFelt(t, "0x1779df1c6de5136ad2620f704b645e9cbd554b57d37f08a06ea60142269c5a5"),
-				ExpectedTxn: InvokeTxnV1example,
+				TxHash:      utils.TestHexToFelt(t, "0xd109474cd037bad60a87ba0ccf3023d5f2d1cd45220c62091d41a614d38eda"),
```

Comment:
For the mock, you'll need to update `mock_starknet_getTransactionByHash` to return the declare transaction

Reply:
Mock tests actually call these functions under the hood